doc/stdenv/stdenv.chapter.md

```diff
@@ -863,9 +871,15 @@ someVar=$(stripHash $name)
 
 ### `wrapProgram` \<executable\> \<makeWrapperArgs\> {#fun-wrapProgram}
 
-Convenience function for `makeWrapper` that automatically creates a sane wrapper file. It takes all the same arguments as `makeWrapper`, except for `--argv0`.
+Convenience function for `makeWrapper` that replaces `<\executable\>` with a wrapper that executes the original program. It takes all the same arguments as `makeWrapper`, except for `--argv0`.
+
+If you will apply it multiple times, it will overwrite the wrapper file and you will end up with double wrapping, which should be avoided.
+
+### `wrapProgramBinary` \<executable\> \<makeBinaryWrapperArgs\> {#fun-wrapProgramBinary}
```

Comment:
nit: This sounds like it wraps a program binary, which I find a pretty confusing name. What about `binaryWrapProgram`?

Reply:
> nit: This sounds like it wraps a program binary, which I find a pretty confusing name. What about `binaryWrapProgram`?

I agree. It's also more consistent with `makeWrapper` vs `makeBinaryWrapper`. I will change it now but this raises the question whether we should make the bash functions have the same name, as @tfc already suggested.

Reply:
Having these functions be drop-in replacements for the shell wrappers would be pretty awesome.

Reply:
i have just tested building and running some more involved applications with a drop-in like `makeWrapper = makeBinaryWrapper` in an overlay and this seems to work perfectly. (only takes some time because you're rebuilding the world afterward)